### PR TITLE
Bug/reply times fix

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR.

**What change(s) does this PR introduce?** 

This PR introduces metric fixes to the `ticket_unassigned_duration_calendar_minutes`, `last_assignment_to_resolution_calendar_minutes`, and `first_reply_time_business_minutes` fields within the `zendesk__ticket_metrics` table.

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [X] No  (please provide explanation how the change is non breaking below.)
This just introduces metric logic changes.

**Is this PR in response to a previously created Issue**

- [X] Yes, Issue #35 
- [ ] No 

**How did you test the PR changes?** 

 
- [X] CircleCi  
- [ ] Other (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 
- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🪑 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/zendesk__ticket_metrics.sql [hunk 1]
```diff
@@ -53,8 +53,14 @@ with ticket_enriched as (
 
 select
   ticket_enriched.*,
-  ticket_reply_times_calendar.first_reply_time_calendar_minutes,
-  ticket_reply_times_calendar.total_reply_time_calendar_minutes,
+  case when coalesce(ticket_comments.count_public_agent_comments, 0) = 0 and ticket_enriched.status = 'solved'
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:56]: tickets that are not solved and have 0 comments would still have a really high `first_reply_time_calendar_minutes` -- why should this depend on the resolution status?

**Review by fivetran-jamie — COMMENTED** [review]
mostly looks good -- some questions around the `first_reply_time_calendar_minutes` stuff

**Review by fivetran-jamie — APPROVED** [review]
gonna approve though since i'll be out next week!

File: models/zendesk__ticket_metrics.sql (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/zendesk__ticket_metrics.sql:56]: Good catch it should not depend on resolution status!

Comment (fivetran-joemarkiewicz) [reply models/zendesk__ticket_metrics.sql:56]: Removed

File: models/intermediate/int_zendesk__updates.sql [hunk 2]
```diff
@@ -28,7 +28,14 @@ with ticket_history as (
         created_at as valid_starting_at,
         lead(created_at) over (partition by ticket_id order by created_at) as valid_ending_at
     from ticket_comment
+
+), final as (
+    select
+        *,
+        first_value(valid_starting_at) over (partition by ticket_id order by valid_starting_at rows unbounded preceding) as ticket_created_date
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/intermediate/int_zendesk__updates.sql]: Good catch @fivetran-jamie I actually just checked and found a few tickets where there was some time between the ticket creation and the first fields being added. I updated to just join from `stg_zendesk__ticket` instead and pull the `created_at`.

**Pull request merged**